assets/src/edit-story/components/canvas/canvasLayout.js

```diff
@@ -37,22 +37,22 @@ const Area = styled.div`
 
 function CanvasLayout() {
 	return (
-		<SelectionCanvas>
-			<Background>
-				<Area area="page">
+		<Background>
+			<Area area="page">
+				<SelectionCanvas>
```

Comment:
@dvoytenko Looks like currently the lasso selection was happening outside of Page, too, including the carousel under the page, creating a selection instead of dragging these. Moved the `SelectionCanvas` around the `Page` only, meaning that now the lasso selection can be drawn within the `Page` only and dragging outside of it won't be handled by lasso selection. Perhaps that's not the best option either though and multi-selecting should be allowed outside of the Page, too? Thoughts?

Reply:
I don't think this is a right approach. I'm pretty certain lasso should work in the canvas. I think it's the other way around instead: carousel should not be part of the canvas. I think logically it'd make more sense: canvas includes the fully drawable area including margin, frame, etc. All tools should be off the canvas. Generally there's very little reason why an inspector is out of canvas but page carousel is on the canvas. We just need to find a way to provide a similar layout with a slightly different layering.

We can, of course, simply rename things and nest them a little different instead of overhauling completely. But I think it's important that the selection area (just like moveable area) geometrically span the whole current canvas area under the current "page", "carousel", "inspector", and "media panel" - we just need to make sure they are not structured in a parent/child relationships so that the event propagation is logical.

In the very short-term, if you don't want to deal with this now, you can just put `onMouseDown={ evt => evt.stopPropagation()}` on the page carousel container. And I can do a pass on the nesting/layering in the meantime so we can discuss it separately?

Reply:
Yep, I basically added this for now just to make it work but didn't think either that this is the right approach. As far as I remember then actually the user should be able to drag elements out of the Page as well, meaning that the selection should work anywhere on the Canvas. Will confirm with Sam just in case but I agree that the carousel should be separated instead of isolating the Page.

I don't mind dealing with this now at all, just wasn't certain of the UX, will take a look tomorrow.

Reply:
There are actually numerous other related bugs with this layout. I reprioritized this a bit on my side. This is still work in progress, but here's what it might involve: https://github.com/ampproject/amp-wp/pull/4110. This PR is pretty close, but couple of issues are still outstanding...

Reply:
@dvoytenko Would you prefer first getting the layers change done and then this PR? Alternatively, we could merge this PR with "incorrect" `SelectionCanvas` and fix it when reorganizing the layers in your PR. LMK.

Note that it looks like lasso selection might not make sense fully outside of the Page since the [elements are deleted](https://www.figma.com/file/Cd9iacVknHK62RO4pRwb68/Stories-Editor?node-id=2832%3A208) when dragged off the Page.

Reply:
@miina I think we'll have to merge this PR first. The layering is a bit more work then we first anticipated. 

> Note that it looks like lasso selection might not make sense fully outside of the Page since the elements are deleted when dragged off the Page.

Yeah, the latest version asks the elements to be deleted when off the screen. But I think the lasso behavior is unchanged. We just want to be able to start the selection anywhere on the workspace (outside the headers, footers, etc). I'll double-check though to be certain.